### PR TITLE
Add breaking changes docs

### LINKED ISSUE #33996: Add the dependent repo and article for breaking changes in the razor compiler
Update the docfx.json and TOC to include the two razor compiler breaking changes articles:

- [Since .NET 8 GA](https://github.com/dotnet/razor/blob/main/docs/Compiler%20Breaking%20Changes%20-%20DotNet%208.md)
- [Since .NET 9 GA](https://github.com/dotnet/razor/blob/main/docs/Compiler%20Breaking%20Changes%20-%20DotNet%209.md)

The titles also need a bit of language changes as well

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #33996

Add the configuration and TOC changes for publishing the breaking changes articles to the razor compiler.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/toc.yml](https://github.com/dotnet/AspNetCore.Docs/blob/774aed08f39a9ba4ef1c586b1ba4ba68f0d2fe0f/aspnetcore/toc.yml) | [aspnetcore/toc](https://review.learn.microsoft.com/en-us/aspnet/core/toc?branch=pr-en-us-34063) |

### CODE REVIEW

File: .openpublishing.publish.config.json [hunk 1]
```diff
@@ -63,6 +63,13 @@
...
       "branch": "main",
       "branch_mapping": {}
+    },
+    {
+      "path_to_root": "_razor",
+      "url": "https://github.com/dotnet/razor",
+      "branch": "main",
+      "include_in_build": true,
+      "branch_mapping": {}
     }
```
Comment (Rick-Anderson) [inline .openpublishing.publish.config.json:73]: That looks correct. See https://github.com/dotnet/AspNetCore.Docs/blob/main/.openpublishing.publish.config.json#L61-L66

File: aspnetcore/toc.yml [hunk 2]
```diff
@@ -797,6 +797,10 @@ items:
             uid: mvc/controllers/filters
           - name: Razor SDK
             uid: razor-pages/sdk
+          - name: Razor breaking changes since .NET 8
+            href: ../_razor/docs/Compiler%20Breaking%20Changes%20-%20DotNet%208.md
+          - name: Razor breaking changes since .NET 9
+            href: ../_razor/docs/Compiler%20Breaking%20Changes%20-%20DotNet%209.md
           - name: View components
```
Comment (Rick-Anderson) [inline aspnetcore/toc.yml:804]: Can you full paths
https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/toc.yml#L1461-L1463

Comment (BillWagner) [reply aspnetcore/toc.yml:804]: The absolute path works, but then the article isn't published in Learn, it's a link.

Here's how it displays for [roslyn breaking changes](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/breaking-changes/compiler%20breaking%20changes%20-%20dotnet%209)

Comment (Rick-Anderson) [reply aspnetcore/toc.yml:804]: > The absolute path works, but then the article isn't published in Learn, it's a link.
> 

The markup for [Sample Docker images](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/toc.yml#L1461-L1463)

which you can see the two TOC entries below [VS tools](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/docker/visual-studio-tools-for-docker). The TOC entries are links that take you out of the aspnet/core doc set.

Can we live with TOC entries that are links?

> Here's how it displays for [roslyn breaking changes](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/breaking-changes/compiler%20breaking%20changes%20-%20dotnet%209)

That's an example from /dotnet, right. And that's how you want it to work, right?

Comment (BillWagner) [reply aspnetcore/toc.yml:804]: > That's an example from /dotnet, right. And that's how you want it to work, right?

Yes, exactly,

**Comment by BillWagner** [comment]
Once dotnet/razor#11199 is merged, this should be fine.

**Comment by BillWagner** [comment]
trying a fresh build

**Pull request merged**